exonum/benches/criterion/block.rs

```diff
@@ -429,31 +430,11 @@ mod foreign_interface_call {
         ) -> Result<(), ExecutionError>;
     }
 
+    #[exonum_interface]
     pub trait ForeignInterface {
         fn timestamp(&self, context: CallContext<'_>, arg: SelfTx) -> Result<(), ExecutionError>;
     }
 
-    impl Interface for dyn ForeignInterface {
-        const INTERFACE_NAME: &'static str = "ForeignInterface";
-
-        fn dispatch(
-            &self,
-            ctx: CallContext<'_>,
-            method: MethodId,
-            payload: &[u8],
-        ) -> Result<(), ExecutionError> {
-            match method {
-                0u32 => {
-                    let bytes = payload.into();
-                    let arg: SelfTx = exonum_merkledb::BinaryValue::from_bytes(bytes)
-                        .map_err(DispatcherError::malformed_arguments)?;
-                    self.timestamp(ctx, arg)
-                }
-                _ => Err(DispatcherError::NoSuchMethod).map_err(From::from),
-            }
-        }
-    }
-
     #[derive(Debug)]
```

Comment:
I implemented this interface by hand to avoid creation `impl Transaction<dyn ForeingInterface> for SelfTx`. 
And I think that we should add attributes to `exonum_interface` to skip creation of this implementation for the specified methods or interfaces.

Reply:
Personally, I think we should refine stubs and use them everywhere (see ECR-3910). 

AFAIK, the earlier definition of `Transaction` featured `Service` as an associated type rather than a type param, hence, using the same transaction argument in several interfaces was *impossible*; now, it's just *difficult*.